packages/editor/src/components/rich-text/style.scss

```diff
@@ -8,6 +8,16 @@
 	margin: 0;
 	position: relative;
 	line-height: $editor-line-height;
+	// In HTML, leading and trailing spaces are not visible, and multiple spaces
+	// elsewhere are visually reduced to one space. This rule prevents spaces
+	// from collapsing so all space is visible in the editor and can be removed.
+	// It also prevents some browsers from inserting non-breaking spaces at the
+	// end of a line to prevent the space from visually disappearing. Sometimes
+	// these non breaking spaces can linger in the editor causing unwanted non
+	// breaking spaces in between words. If also prevent Firefox from inserting
+	// a trailing `br` node to visualise any trailing space, causing the element
+	// to be saved.
+	white-space: pre-wrap;
```

Comment:
`pre-wrap` will preserve newlines, spaces, and tabs. But in `filterStringComplete` we're only replacing newlines and tabs with spaces. As a consequence of this change, the editor will now show those consecutive spaces.

Before this change:

![peek 2018-11-27 19-21](https://user-images.githubusercontent.com/583546/49103247-19a5de00-f27b-11e8-8b6f-8c19688269cb.gif)

After this change:

![peek 2018-11-27 19-29](https://user-images.githubusercontent.com/583546/49103285-304c3500-f27b-11e8-8751-b4dbb70e301b.gif)

Is that a concern?

Reply:
For completeness, here's the same action in visual mode. No visual change, although under the hood the HTML doesn't add `nbsp`.

Before this change:

![peek 2018-11-27 19-23](https://user-images.githubusercontent.com/583546/49103574-00516180-f27c-11e8-8c2d-3f6f46db4925.gif)

After this change:

![peek 2018-11-27 19-28](https://user-images.githubusercontent.com/583546/49103582-06dfd900-f27c-11e8-8613-4161d77fc045.gif)

I've noticed that with this change applied the leading spaces in the first line collapse before the post is reloaded.

Reply:
I think that’s fine?

Reply:
In the rare case that the user want to have those spaces visible on the front end, I think they should insert those non breaking spaces themselves.